### PR TITLE
Feat(web-react): Uncontrolled Pagination component

### PR DESCRIPTION
**Pull request by literat**
## Description

I am introducing a new Pagination component.

### Additional context

Based on https://github.com/lmc-eu/spirit-design-system/pull/738

### Issue reference

---

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md).
- [ ] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/web-react/src/components/Pagination/README.md [hunk 1]
```diff
@@ -132,6 +144,16 @@ This component extends the `PaginationButtonLink` component.
 | -------------------- | -------- | ------- | -------- | ------------------------------- |
 | `accessibilityLabel` | `string` | `Next`  | ✕        | Accessibility label of the link |
 
+## UncontrolledPagination props
+
+| Name                 | Type                           | Default | Required | Description                                |
+| -------------------- | ------------------------------ | ------- | -------- | ------------------------------------------ |
+| `accessibilityLabel` | `string`                       | ✕       | ✕        | Accessibility label of the link            |
+| `defaultPage`        | `number`                       | `1`     | ✕        | Page number of default page                |
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/README.md:152]: i don't understand this prop. What does it do?

File: packages/web-react/src/components/Pagination/README.md [hunk 2]
```diff
@@ -132,6 +144,16 @@ This component extends the `PaginationButtonLink` component.
 | -------------------- | -------- | ------- | -------- | ------------------------------- |
 | `accessibilityLabel` | `string` | `Next`  | ✕        | Accessibility label of the link |
 
+## UncontrolledPagination props
+
+| Name                 | Type                           | Default | Required | Description                                |
+| -------------------- | ------------------------------ | ------- | -------- | ------------------------------------------ |
+| `accessibilityLabel` | `string`                       | ✕       | ✕        | Accessibility label of the link            |
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/README.md:151]: what link? The arrows or the numbers? Am I able to set the a11y label of arrows?

File: packages/web-react/src/components/Pagination/README.md [hunk 3]
```diff
@@ -132,6 +144,16 @@ This component extends the `PaginationButtonLink` component.
...
+| `defaultPage`        | `number`                       | `1`     | ✕        | Page number of default page                |
+| `chapterSize`        | `number`                       | `5`     | ✕        | Number of displayed pages between chevrons |
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/README.md:153]: how about page count or something like that? I am not sure if the keyword chapter is the right one.

File: packages/web-react/src/components/Pagination/README.md [hunk 4]
```diff
@@ -132,6 +144,16 @@ This component extends the `PaginationButtonLink` component.
...
+| `onChange`           | `(pageNumber: number) => void` | ✕       | ✕        | On page change callback                    |
+| `totalNumber`        | `number`                       | -       | ✕        | Total count of pages                       |
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/README.md:155]: last page?

File: packages/web-react/src/components/Pagination/__tests__/UncontrolledPagination.test.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+describe('UncontrolledPagination', () => {
+  const onPageChange = jest.fn();
+
+  beforeEach(() => {
+    onPageChange.mockClear();
+  });
+
+  it('renders pagination items with current page selected', () => {
+    render(<UncontrolledPagination totalNumber={10} defaultPage={5} onChange={onPageChange} />);
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/__tests__/UncontrolledPagination.test.tsx:13]: so the default page should be currentPage?

File: packages/web-react/src/components/Pagination/README.md (hunk 2 above)
Comment (literat) [reply packages/web-react/src/components/Pagination/README.md:151]: of the previous and next link...

**Comment by literat** [comment]
Discussed the suggestion offline with @crishpeen and refactored it.

File: packages/web-react/src/components/Pagination/README.md [hunk 6]
```diff
@@ -132,6 +144,18 @@ This component extends the `PaginationButtonLink` component.
...
 
+## UncontrolledPagination props
+
+| Name                         | Type                           | Default    | Required | Description                                                         |
+| ---------------------------- | ------------------------------ | ---------- | -------- | ------------------------------------------------------------------- |
+| `accessibilityLabel`         | `string`                       | ✕          | ✕        | Accessibility label of the page links                               |
+| `accessibilityLabelNext`     | `string`                       | `Next`     | ✕        | Accessibility label of the next link                                |
+| `accessibilityLabelPrevious` | `string`                       | `Previous` | ✕        | Accessibility label of the previous link                            |
+| `defaultPage`                | `number`                       | `1`        | ✕        | The number of the page selected as current page at the first render |
+| `onChange`                   | `(pageNumber: number) => void` | ✕          | ✕        | On page change callback                                             |
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/README.md:155]:
```suggestion
| `onChange`                   | `(pageNumber: number) => void` | —          | ✕        | On page change callback                                             |
```

File: packages/web-react/src/components/Pagination/README.md [hunk 7]
```diff
@@ -132,6 +144,18 @@ This component extends the `PaginationButtonLink` component.
 | -------------------- | -------- | ------- | -------- | ------------------------------- |
 | `accessibilityLabel` | `string` | `Next`  | ✕        | Accessibility label of the link |
 
+## UncontrolledPagination props
+
+| Name                         | Type                           | Default    | Required | Description                                                         |
+| ---------------------------- | ------------------------------ | ---------- | -------- | ------------------------------------------------------------------- |
+| `accessibilityLabel`         | `string`                       | ✕          | ✕        | Accessibility label of the page links                               |
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/README.md:151]:
```suggestion
| `accessibilityLabel`         | `string`                       | —          | ✕        | Accessibility label of the page links                               |
```

File: packages/web-react/src/components/Pagination/README.md [hunk 8]
```diff
@@ -132,6 +144,18 @@ This component extends the `PaginationButtonLink` component.
...
+| `accessibilityLabelNext`     | `string`                       | `Next`     | ✕        | Accessibility label of the next link                                |
+| `accessibilityLabelPrevious` | `string`                       | `Previous` | ✕        | Accessibility label of the previous link                            |
+| `defaultPage`                | `number`                       | `1`        | ✕        | The number of the page selected as current page at the first render |
+| `onChange`                   | `(pageNumber: number) => void` | ✕          | ✕        | On page change callback                                             |
+| `totalPages`                 | `number`                       | -          | ✕        | Total count of pages                                                |
+| `visiblePages`               | `number`                       | `5`        | ✕        | Number of displayed pages between chevrons                          |
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/README.md:157]:
```suggestion
| `visiblePages`               | `number`                       | `5`        | ✕        | Number of displayed pages                        |
```

There is only one chevron when you are on the edge. Maybe we can say `numbered pages` or we can keep it as i suggest here ^

File: packages/web-react/src/components/Pagination/UncontrolledPagination.tsx [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+import PaginationLinkNext from './PaginationLinkNext';
+import PaginationLinkPrevious from './PaginationLinkPrevious';
+import { usePagination } from './usePagination';
+
+export const UncontrolledPagination = (props: SpiritUncontrolledPaginationProps): JSX.Element => {
+  const {
+    accessibilityLabel,
+    accessibilityLabelPrevious = 'Previous',
+    accessibilityLabelNext = 'Next',
+    totalPages = 0,
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/UncontrolledPagination.tsx:15]: in docs we say there is no default

File: packages/web-react/src/components/Pagination/UncontrolledPagination.tsx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    defaultPage,
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/UncontrolledPagination.tsx:16]: this has default value in docs

File: packages/web-react/src/components/Pagination/UncontrolledPagination.tsx (hunk 9 above)
Comment (crishpeen) [reply packages/web-react/src/components/Pagination/UncontrolledPagination.tsx:15]: abc sort please

File: packages/web-react/src/components/Pagination/__tests__/UncontrolledPagination.test.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+  it('renders pagination items with current page selected', () => {
+    render(
+      <UncontrolledPagination accessibilityLabel="test page" totalPages={10} defaultPage={5} onChange={onPageChange} />,
+    );
+
+    const items = screen.getAllByRole('button');
+    expect(items).toHaveLength(7);
+
+    const selectedPageItem = screen.getByLabelText('Current Page, Page 5');
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/__tests__/UncontrolledPagination.test.tsx:20]: shouldn't `Current Page, Page 5` be overridden by `test page`?

File: packages/web-react/src/components/Pagination/stories/UncontrolledPagination.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    defaultPage: {
+      control: 'number',
+      description: 'Default page for the first render; please reload the page to apply this setting.',
+      table: {
+        defaultValue: { summary: 1 },
+      },
+    },
+    totalPages: {
+      control: 'number',
+      table: {
```
Comment (crishpeen) [inline packages/web-react/src/components/Pagination/stories/UncontrolledPagination.stories.tsx:22]: you can remove this. There is no default and you set the prop in the `args` section.

File: packages/web-react/src/types/pagination.ts [hunk 13]
```diff
@@ -44,3 +48,21 @@ export type SpiritPaginationLinkProps<E extends ElementType = 'a'> = PaginationL
...
 
 export type SpiritPaginationLinkPreviousNextProps<E extends ElementType = 'a'> = PaginationLinkPreviousNextProps<E>;
+
+export interface UncontrolledPaginationProps {
+  accessibilityLabelNext?: string;
+  accessibilityLabelPrevious?: string;
+  defaultPage: number;
+  visiblePages?: number;
+  onChange: (pageNumber: number) => void;
+  totalPages: number;
```
Comment (crishpeen) [inline packages/web-react/src/types/pagination.ts:58]: in the api docs nothing is required, but here defaultPage, onChange and totalPages are required

**Review by crishpeen — APPROVED** [review]

**Pull request merged**